### PR TITLE
Release 2.8.0

### PR DESCRIPTION
**Pull request by hpidcock**
- Update facade methods for Juju 2.8.0
- Fixes codegen for Python 3.7+
- Nested facade definitions are now deserialised properly (e.g. storage on ApplicationDeploy)

### CODE REVIEW

File: juju/client/_client10.py [hunk 1]
```diff
@@ -996,7 +996,7 @@ async def Get(self, application=None, branch=None):
         '''
         application : str
         branch : str
-        Returns -> typing.Union[str, typing.Mapping[str, typing.Any], _ForwardRef('Value')]
+        Returns -> typing.Union[str, typing.Mapping[str, typing.Any], ForwardRef('Value')]
```
Comment (howbazaar) [inline juju/client/_client10.py:999]: Do you know why these changed?

Comment (hpidcock) [reply juju/client/_client10.py:999]: They were internal types pre 3.7

**Comment by hpidcock** [comment]
> Good catch on the nested args.
> I assume there's been a manual test done using this to drive a 2.8 controller. If not, can it be done prior to landing?

Yep testing some examples against both a k8s and lxd controller on 2.8

**Comment by hpidcock** [comment]
!!build!!

**Comment by hpidcock** [comment]
$$merge$$

**Comment by hpidcock** [comment]
$$merge$$

**Comment by ryan-beisner** [comment]
Hi all - this patch and the 2.8.0 libjuju release regressed usage in all of the OpenStack Charm gates:
https://github.com/juju/python-libjuju/issues/413

**Pull request merged**